assets/filter-products.js

```diff
@@ -1,209 +1,184 @@
-let currentCursor = null;
-let hasNextPage = false;
-let hasPrevPage = false;
-let prevCursors = [];
-let prevButton = document.getElementById('prev-button');
-let nextButton = document.getElementById('next-button');
-let mainCollectionGrid = document.querySelector('.main-collection-grid-wrapper');
-const parts = window.location.pathname.split('/').filter(part => part.length > 0);
-const collectionHandle = parts.length > 0 ? parts.pop() : 'hydrogen';
-let filterOptionPrice = document.querySelectorAll('.js-filter-price input');
-let filterOptionList = document.querySelectorAll('input[data-filter-type]');
-let filterOptionQuery = '';
-
-async function fetchFilteredProducts (filterOptionListQuery, minPrice, maxPrice) {
-  let afterPart = prevCursors.length > 1 ? `, after: \"${currentCursor}\"` : '';
-  let filterListOptionQuery = filterOptionListQuery.length > 0 ? filterOptionListQuery : '';
-  let filters = `[
-  ${filterListOptionQuery}
-  { price: { min: ${minPrice}, max: ${maxPrice} } }]`;
-
-  let query = JSON.stringify({
-    query: `{
-      collection(handle: "${collectionHandle}") {
-        handle
-        products(first: 5${afterPart}, filters: ${filters}) {
-        filters {
-        id
-        label
-        type
-        values {
-          id
-          label
-          count
-          input
-        }
-      }
-        edges {
-          node {
-            handle
-            availableForSale
-            productType
-            vendor
-            tags
-            variants(first: 10) {
-              edges {
-                node {
-                  id
-                  title
-                  selectedOptions {
-                    name
-                    value
-                  }
-                }
-              }
-            }
-          }
-        }
-          pageInfo {
-            hasNextPage
-            hasPreviousPage
-            endCursor
-            startCursor
-          }
+import { generateQuery } from './query.js';
+class filterComponent extends HTMLElement {
+  constructor() {
+    super();
+    this.currentCursor;
+    this.hasNextPage;
+    this.hasPrevPage;
+    this.prevCursors = [];
+    this.prevButton = this.querySelector('#prev-button');
```

Comment:
Please move selectors into `selectors` object and use them in every querySelector